lib/luarocks-version.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * Utilities relating to Lua 'rocks' version numbers.
+ * This compares version numbers using the algorithm
+ * followed by luarocks command-line utility
+ */
+'use strict';
+
+// Compare two arrays containing splitted and transformed to
+// positive/negative numbers parts of version strings,
+// respecting negative/missing values:
+// [1, 2, 1] > [1, 2], but [1, 2, -1] < [1, 2] ([1, 2] is aligned to [1, 2, 0])
```

Comment:
Would it be worth adding a unit test with these examples?

Reply:
You're absolutely right! I've added unit tests… and found a bug! :)